modules/scaleform.lua

```diff
@@ -0,0 +1,224 @@
+        end
+    end
+end
+
+local function convertArgs(argsTable)
+    for i=1, #argsTable do -- loop through the args
+        local arg = argsTable[i] -- Set the value to the current arg
+        if type(arg) == 'string' then -- If the type of v is a string
+            ScaleformMovieMethodAddParamPlayerNameString(arg) -- Add the player name string
+        elseif type(arg) == 'number' then -- If the type of v is a number
+            if math.type(arg) == 'integer' then -- If the math type of v is an integer
+                ScaleformMovieMethodAddParamInt(arg) -- Add the integer
+            else -- If the math type of v is not an integer
+                ScaleformMovieMethodAddParamFloat(arg) -- Add the float
+            end
+        elseif type(arg) == 'boolean' then -- If the type of v is a boolean
+            ScaleformMovieMethodAddParamBool(arg) -- Add the boolean
+        else
+            error(('Unsupported Parameter type [%s]'):format(type(arg))) -- Error unsupported type
+        end
+    end
+end
+
+---@param type string
+---@return boolean | int | string
+---@description Awaits the return value, and converts it to a usable data type
+local function retrieveReturnValue(type)
+    local result = EndScaleformMovieMethodReturnValue() -- End the scaleform movie method with a return value
+
+    lib.waitFor(function()
+        if IsScaleformMovieMethodReturnValueReady(result) then
+            return true
+        end
+    end, "Failed to retrieve return value", 1000)
+
+    if type == "int" then -- If the type is an integer
+        return GetScaleformMovieMethodReturnValueInt(result) -- Get the return value as an integer
+    elseif type == "bool" then
+        return GetScaleformMovieMethodReturnValueBool(result) -- Get the return value as a boolean
+    else -- If the type is not an integer
+        return GetScaleformMovieMethodReturnValueString(result) -- Get the return value as a string
+    end
+end
+
+---@param name string
+---@param args? table
+---@param returnValue? string
+---@return any
+---@description Call a scaleform function, with optional args or return value.
+function Scaleform:method(name, args, returnValue)
+    if not self.handle then -- If the scaleform handle is nil
+        return error('Scaleform handle is nil') -- Error the scaleform handle is nil
+    end
+
+    if args and type(args) ~= 'table' then -- If the type of args is not a table
+        return error('Args must be a table') -- Error args must be a table
+    end
+
+    BeginScaleformMovieMethod(self.handle, name) -- Begin the scaleform movie method
+
+    -- Converts the arguments into data types usable by scaleform
+    if args then
+        convertArgs(args)
+    end
+
+    -- When wanting a return value, the scaleform has to be ended with a different native
+    if returnValue then
+        retrieveReturnValue(returnValue)
+    else
+        EndScaleformMovieMethod() -- End the scaleform movie method
+    end
```

Comment:
```suggestion
    if returnValue then
        return retrieveReturnValue(returnValue)
    end

    EndScaleformMovieMethod() -- End the scaleform movie method
```
Let's actually return the retrieved return value.